tests/Omnipay/Common/Message/AbstractRequestTest.php

```diff
@@ -87,6 +101,17 @@ public function testAmountWithEmpty()
         $this->assertSame(null, $this->request->getAmount());
     }
 
+    /**
+     * @expectedException Omnipay\Common\Exception\InvalidRequestException
+     * @expectedExceptionMessage A zero amount is not allowed.
+     */
+    public function testAmountZeroNotAllowed()
+    {
+        $this->changeProtectedProperty('zeroAmountAllowed', false);
+        $this->request->setAmount('0.00');
+        $this->request->getAmount();
+    }
+
```

Comment:
This is correct, however, when I was looking through the test results, I saw that this particular Exception being thrown as not being tested, so I added it to this branch.